for editing (#2)
DevExpress-Examples/web-forms-dashboard-save-dashboards-to-database: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T386418)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->
#  Dashboard for Web Forms - How to Load and Save Dashboards from/to a Database

> This example shows how to create a custom dashboard storage that allows you to store dashboards in a database.

Reviewer: ` custom dashboard storage` already means `store dashboards`

Author: ```suggestion
This example shows how to create a custom dashboard storage to load and save dashboards in a database.
```

<!-- default file list -->
## Files to Look At

* [DataBaseEditaleDashboardStorage.cs](./CS/DataBaseEditaleDashboardStorage.cs) (VB: [DataBaseEditaleDashboardStorage.vb](./VB/DataBaseEditaleDashboardStorage.vb))
* [Global.asax.cs](./CS/Global.asax.cs) (VB: [Global.asax.vb](./VB/Global.asax.vb))
<!-- default file list end -->

## Example Overview

> The example uses the [System.Data.SqlClient](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient?redirectedfrom=MSDN&view=net-5.0) members to connect and operate an MS SQL server database.

Reviewer: Why do you use `operate` here? What does this imply?

Author: ```suggestion
The example uses the [System.Data.SqlClient](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient?redirectedfrom=MSDN&view=net-5.0) members to connect and manage an MS SQL server database.
```

A custom dashboard storage should implement one of the following interfaces: [IDashboardStorage](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IDashboardStorage) or [IEditableDashboardStorage](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IEditableDashboardStorage).

> The following API used in the example:

Reviewer: ```suggestion
The following API is used in the example:
```

- [LoadDashboard](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IDashboardStorage.LoadDashboard(System.String)) 

>     Returns a dashboard by its ID in the XDocument format, which describes an object model of the dashboard.

Reviewer: - `Returns a dashboard by its ID` Do you mean 'Returns a dashboard with the corresponding ID'?
- `in the XDocument format` -> in XDocument format
- `which describes` That does this refer to? The dashboard or the format? 'which' is covered in our guidelines

Author: ```suggestion
    Loads a dashboard with the specified ID in XDocument format from the storage. 
```

Reviewer: Loads a dashboard with the specified ID in XDocument format from storage.

Author: ```suggestion
    Loads a dashboard with the specified ID in XDocument format from storage.
```

- [GetAvailableDashboardsInfo](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IDashboardStorage.GetAvailableDashboardsInfo) 

    Returns a list of IDs and Captions of dashboards available in the data storage.
- [SaveDashboard](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IDashboardStorage.SaveDashboard(System.String-System.Xml.Linq.XDocument)) 

>     Updates the dashboard with new settings by its id.

Reviewer: - `with new settings by its id` Again, why do you use 'by something'? This does not make sense
- `id` Should this not be ID?

Author: ```suggestion
    Saves the specified dashboard with new settings to the dashboard storage.
```

Reviewer: Saves the specified dashboard with new settings to dashboard storage

- [AddDashboard](https://docs.devexpress.com/Dashboard/DevExpress.DashboardWeb.IEditableDashboardStorage.AddDashboard(System.Xml.Linq.XDocument-System.String)) 

>    Takes a dashboard definition with its caption, saves it to the data storage, and returns the ID of a new saved dashboard.

Reviewer: ```suggestion
   Saves a dashboard definition and its caption to the data storage and returns the ID of the new saved dashboard.
```

> Additionally, this example contains an SQL file ([SavedDashboards.sql](./CS/SavedDashboards.sql)), which can be used to recreate a database used in this example on your side. Do no forget to update the connection string in the **Web.config** file to make it valid in your environment.

Reviewer: - Again `which`
- `on your side` means what?
- Why do you use `Do no forget`?

Author: ```suggestion
Additionally, this example contains an SQL file ([SavedDashboards.sql](./CS/SavedDashboards.sql)) that can be used to recreate a database used in this example. Update the connection string in the **Web.config** file to make it valid in your environment.
```
